frontend/src/metabase/common/components/DataPicker/hooks/useAutoSelectOnlyItem.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from "react";
+import { useLatest } from "react-use";
+
+/**
+ * Automatically selects the only item on the list.
+ * Does nothing if there's 0 items or more than 1.
+ *
+ * @returns true when there's only 1 item.
+ */
+export const useAutoSelectOnlyItem = <Item>(
+  items: Item[] | undefined,
```

Comment:
It's easier to handle `undefined` here.

----

If we do this:

```suggestion
  items: Item[] = [],
```

then the effect will run on every render.

----

If we do this:

```suggestion
  items: Item[] = EMPTY_ARRAY,
```

then it's impossible to type `EMPTY_ARRAY` due to this hook having a generic interface.

----

If we do this:

```suggestion
  items: Item[] = getProperlyTypedEmptyArray(),
```

then it's an overkill. 

----

If we provide the default value from outside, then we pollute all usages of this hook with that logic.
And it does not fit really well there, because `ItemList` component relies on `items` being `undefined` and not an empty array.